beancount_import/source/generic_importer_source.py

```diff
@@ -0,0 +1,188 @@
+                                               self.comparator))
+            # If journal has n cleared entries for this hash, pick remaining
+            for entry in entries[hash_][n:]:
+                # add importer name as sorce description to source postings
+                self._add_description(entry)
+                # balance amount
+                self.balance_amounts(entry)
+                uncleared_entries[hash_].append(entry)
+
+        results.add_pending_entries(
+            [ImportResult(entry.date, [entry], None)
+                for entry in itertools.chain.from_iterable(uncleared_entries.values())
+            ]
+        )
+
+    def comparator(self, entry1, entry2):
+        """Returns if the two entries are similar and 2nd entry is cleared.
+        The first entry is from new_entries and 2nd is from journal
+        """
+        return self._comparator(entry1, entry2) \
+               and self.is_entry_cleared(entry2) \
+               and entry1.narration == entry2.postings[0].meta['source_desc']
+
+    def _add_description(self, entry: Transaction):
+        if not isinstance(entry, Transaction): return None
+        postings = entry.postings #type: ignore
+        to_mutate = []
+        for i, posting in enumerate(postings):
+            if posting.account != self.account: continue
+            if isinstance(posting.meta, dict): posting.meta["source_desc"] = entry.narration
+            else: to_mutate.append(i)
+        for i in to_mutate:
+            p = postings.pop(i)
+            p = Posting(p.account, p.units, p.cost, p.price, p.flag, {"source_desc":entry.narration})
+            postings.insert(i, p)
+
+    @staticmethod
+    def balance_amounts(txn:Transaction)-> None:
+        """Add FIXME account for the remaing amount to balance accounts"""
+        inventory = SimpleInventory()
+        for posting in txn.postings:
+            inventory += posting.units
+        for currency in inventory:
+            txn.postings.append(
+                Posting(
+                    account=FIXME_ACCOUNT,
+                    units=Amount(currency=currency, number=-inventory[currency]),
+                    cost=None,
+                    price=None,
+                    flag=None,
+                    meta={},
+                ))
+
+    @staticmethod
+    def _hash_entry(entry:Directive, exclude_meta_keys=frozenset()) -> str:
+        """Similar to beancount.core.compare.hash_entry but can skip selective meta fields
+        the meta fields to be used for hashing should be in Transaction's meta, not Posting's meta
+        """
+        if not isinstance(entry, Transaction): return hash_entry(entry)
+        h = hashlib.md5()
+        h.update(hash_entry(entry, exclude_meta=True).encode())
+        for key in entry.meta:
+            if key in exclude_meta_keys: continue
+            h.update(str(entry.meta[key]).encode())
+        return h.hexdigest()
+
+    def is_entry_cleared(self, entry: Transaction) -> bool:
```

Comment:
If you use get_pending_and_invalid_entries from description_based_source.py, you can also just inherit the is_posting_cleared definition from description_based_source.py and not have to define either is_entry_cleared or is_posting_cleared here.

Reply:
Done! I was actually calling `super().is_posting_cleared` after checking `entry.account == self.account`, but I guess that is no more needed since I removed `is_entry_cleared` where I was calling this method for all postings.  
removed.